react/week4/src/components/FetchUser.js

```diff
@@ -0,0 +1,44 @@
+import React,{useState,useEffect,createContext} from 'react';
+import SearchForm from './SearchForm';
+import UserList from "./UserList";
+
+export const UserContext = createContext();
+
+  const FetchUser = () =>{
+    const [users, setUsers] = useState([]);
+    const [error, setError] = useState(null);
+    const [isLoading, setIsLoading] = useState(false);
+   const [query, setQuery] = useState();
+    
+    useEffect(() => {
+        async function fetchData() {
+            setIsLoading(true);
+            try {
+                console.log(query);
+                const response = await fetch(`https://api.github.com/search/users?q=${query}`)
+                const data = await response.json();
+                console.log(data);
+                setUsers(data.items);
+                console.log(data.items);
+                setIsLoading(false);
+            } catch (error) {
+                setError(error);
+                }
+            };
```

Comment:
Set `isLoading=false` inside a `finally {}` block, this way, even if the requests succeeds or fails, the `isloading` will be set to false. Currently, it will only be set if it succeeds.
```suggestion
            try {
                console.log(query);
                const response = await fetch(`https://api.github.com/search/users?q=${query}`)
                const data = await response.json();
                console.log(data);
                setUsers(data.items);
                console.log(data.items);
            } catch (error) {
                setError(error);
            } finally {
                setIsLoading(false);
            }
        };
```